### DISCUSSION TITLE
Simulations

### DISCUSSION THREAD

**Discussion by ChrisKonig** [LAPKB/Pmetrics#232] [Q&A] [answered]
Hi,
I am still new to this. I am currently simulating new dosages based on the ind popPK estimates of my individuals. 
Is there a way to do that more smoothly than I do? 
Currently I am adding this type of code for every ID with manually adding the pk estimates: 

_#patient 3 ED 2.3 sim sim 1g q8h 0.5h inf
SIMrun(poppar=list(weight=1, mean= c(2.756,44.714, 7.3554, 19.276), covar= m), limits=NA, model="1model.txt",
       data="simdata.csv", nsim=1, predInt = c(24,36,0.25))
SIMdata <- SIMparse(file="simout?.txt")
SIMparse(file="simout?.txt")_

When running this with different IDs one after another PMetrics asks me: 

_The working directory already contains 1 files matching the output name (from Wed Jun 19 2024).
What would you like to do?
1) delete all existing files 'simout*.txt'
2) prefix existing files with 'old_' (overwriting any that may already exist)
3) abort function_

Which makes totally sense and I am fine with deleting the simout file. 
I assume there is a more elegant way to do that. 
Is there a code I can use within the simulation code so I don't have to manually confirm "deleting"? 
Also, can I provide multiple pk estimates as a csv file? And execute the simulations in separate csv thereafter? 

I truly appreciate insights in how to do that more efficiently as I have to run multiple different simulations. 

Thanks, Chris

**Comment by mnneely** [comment, marked as answer]
Just add each "patient" template to the same data.csv file. First patient becomes subject ID 1, next one is ID 2, etc. It looks just like a data file for a popPK study, except all the OUT values are replaced by simulated ones. That's why I put -1 as the value for OUT in simulation templates. Of course, there are probably fewer observation times in a simulation template than a true data file because `predInt` can fill in the gaps.

If you want to use the posterior values to simulate, then make sure you have matching IDs and the number of subjects in the template as you did in the pop model.  Then say `usePost = TRUE` when you simulate. This uses the Bayesian posterior distribution of subject 1 in the pop model and the data template for subject 1. Then posterior from subject 2 and template 2, etc.
